components/lib/crash/src/main/java/mozilla/components/lib/crash/handler/CrashHandlerService.kt

```diff
@@ -4,42 +4,57 @@
 
 package mozilla.components.lib.crash.handler
 
-import android.app.IntentService
+import android.app.Service
 import android.content.Intent
-import mozilla.components.lib.crash.CrashReporter
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
 import mozilla.components.lib.crash.Crash
+import mozilla.components.lib.crash.CrashReporter
+import mozilla.components.lib.crash.R
+import mozilla.components.lib.crash.notification.CrashNotification
+import mozilla.components.support.base.ids.SharedIdsHelper
 
-private const val WORKER_THREAD_NAME = "CrashHandlerService"
+private const val NOTIFICATION_TAG = "mozac.lib.crash.handlecrash"
+private const val NOTIFICATION_ID = 1
 
 /**
  * Service receiving native code crashes (from GeckoView).
  */
-class CrashHandlerService : IntentService(WORKER_THREAD_NAME) {
-    private val logger by lazy { CrashReporter
-        .requireInstance
-        .logger
-    }
-
-    public override fun onHandleIntent(intent: Intent?) {
-        if (intent == null) {
-            return
+class CrashHandlerService : Service() {
+    private val crashReporter: CrashReporter by lazy { CrashReporter.requireInstance }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        crashReporter.logger.error("CrashHandlerService received native code crash")
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = CrashNotification.ensureChannelExists(this)
+            val notification = NotificationCompat.Builder(this, channel)
+                .setContentTitle(getString(R.string.mozac_lib_crash_dialog_title,
+                    crashReporter.promptConfiguration.organizationName))
+                .setSmallIcon(R.drawable.mozac_lib_crash_notification)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setCategory(NotificationCompat.CATEGORY_ERROR)
+                .setAutoCancel(true)
+                .build()
+
+            val notificationId = SharedIdsHelper.getIdForTag(this, NOTIFICATION_TAG)
+            startForeground(notificationId, notification)
         }
 
-        logger.error("CrashHandlerService received native code crash")
-
+        NotificationManagerCompat.from(this).cancel(NOTIFICATION_TAG, NOTIFICATION_ID)
         intent.extras?.let { extras ->
             val crash = Crash.NativeCodeCrash.fromBundle(extras)
+            CrashReporter.requireInstance.onCrash(this, crash)
+        } ?: crashReporter.logger.error("Received intent with null extras")
```

Comment:
`IntentService` did run this on a background thread and now we are running this on the main thread. Could this be a problem? Are we potentially blocking the main thread here?

Reply:
We are probably blocking the main thread here.  I didn't think about this case since by the time we're here there's no UI thread.  I'll look into this and see if we should not run this in the main thread.

Reply:
Agreed, I'll move this into the IO thread.